src/modules/c.rs

```diff
@@ -65,12 +65,12 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
                     // so again we always want the first semver-ish word.
                     VersionFormatter::format_module_version(
                         module.get_name(),
-                        c_compiler_info.split_whitespace().find_map(
-                            |word| match Version::parse(word) {
-                                Ok(_v) => Some(word),
-                                Err(_e) => None,
-                            },
-                        )?,
+                        c_compiler_info.split_whitespace().find(|word| {
+                            match Version::parse(word) {
+                                Ok(_v) => true,
+                                Err(_e) => false,
+                            }
+                        })?,
```

Comment:
#4002 has a fix for this.

Reply:
Should I wait for it to be merged and rebase on top of it?

Reply:
Changes reverted by https://github.com/starship/starship/pull/4004/commits/da426f34b864b1e17636da64083c57f3b33d874e. I'll wait for #4002 to be merged and rebase on top of it, thanks for letting me know.